docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 7aff6a6..09d640b 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -24,7 +24,7 @@ mode="dpdk"
+ # Gateway interface(s)
+ #
+ # In the order of ("s1u" "sgi")
+-ifaces=("ens803f2" "ens803f3")
++ifaces=("enp134s0" "enp136s0")
+
+ # Static IP addresses of gateway interface(s) in cidr format
+ #
+@@ -34,7 +34,7 @@ ipaddrs=(198.18.0.1/30 198.19.0.1/30)
+ # MAC addresses of gateway interface(s)
+ #
+ # In the order of (s1u sgi)
+-macaddrs=(9e:b2:d3:34:ab:27 c2:9c:55:d4:8a:f6)
++macaddrs=(40:a6:b7:78:3f:ec 40:a6:b7:78:3f:e8)
+
+ # Static IP addresses of the neighbors of gateway interface(s)
+ #
+@@ -44,7 +44,7 @@ nhipaddrs=(198.18.0.2 198.19.0.2)
+ # Static MAC addresses of the neighbors of gateway interface(s)
+ #
+ # In the order of (n-s1u n-sgi)
+-nhmacaddrs=(22:53:7a:15:58:50 22:53:7a:15:58:50)
++nhmacaddrs=(40:a6:b7:78:3f:bc 40:a6:b7:78:3f:b8)
+```
+
+6) Modify the script [docker_setup.sh](./scripts/docker_setup.sh) and update the function `move_ifaces()` in condition `if [ "$mode" == 'cndp' ]` . Update `start_q_idx` to choose the start queue index to receive n/w packets. This should match the queue id used in lports section of [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc).To get better performance (optional step), assign `cpuset-cpus` in [docker_setup.sh](./scripts/docker_setup.sh) to cores ids same as queue ids used to receive n/w packets.
+
+```
+diff --git a/conf/cndp_upf_1worker.jsonc b/conf/cndp_upf_1worker.jsonc
+index da60d51..b1c3df6 100644
+--- a/conf/cndp_upf_1worker.jsonc
++++ b/conf/cndp_upf_1worker.jsonc
+@@ -86,7 +86,7 @@
+     "lports": {
+         "enp134s0:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 6,
+             "umem": "umem0",
+             "region": 0,
+             "busy_poll": true,
+@@ -96,7 +96,7 @@
+         },
+         "enp136s0:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 6,
+             "umem": "umem0",
+             "region": 1,
+             "busy_poll": true,
+```
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 9058839..2e4e505 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -104,7 +104,7 @@ function move_ifaces() {
+                        # num queues
+                        num_q=1
+                        # start queue index
+-                       start_q_idx=22
++                       start_q_idx=6
+                        # RSS using TC filter
+                        setup_tc "${ifaces[$i]}" $num_q $start_q_idx
+                fi
+@@ -218,7 +218,7 @@ fi
+ # Run bessd
+ docker run --name bess -td --restart unless-stopped \
+-       --cpuset-cpus=12-13 \
++       --cpuset-cpus=6-7 \
+        --ulimit memlock=-1 -v /dev/hugepages:/dev/hugepages \
+        -v "$PWD/conf":/opt/bess/bessctl/conf \
+        --net container:pause \
+
+```
```

Comment:
Why are not these changes made as part of step 3)?

Reply:
These changes are part of step 6. I prefer to describe changes in different steps rather than have a long step 3.

Reply:
Ok